Fixed issue with HTML attachments (501 Syntax error - line too long)

sanzaru opened the pull request:
The PR fixes an issue that occurred when trying to send long HTML attachments. The server then responded with an "501 Syntax error - line too long" error.
I also fixed some closures.

## Motivation and Context
I wanted to send some HTML mails and ran into the error, also it is subject of the following issue: [https://github.com/Kitura/Swift-SMTP/issues/106](https://github.com/Kitura/Swift-SMTP/issues/106)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Kitura/Swift-SMTP?pullRequest=114) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Kitura/Swift-SMTP?pullRequest=114) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Kitura/Swift-SMTP?pullRequest=114) it.</sub>

dannys42 commented on Sources/SwiftSMTP/SMTP.swift:
```diff
@@ -99,13 +99,13 @@ public struct SMTP {
     ///     - completion: Callback when sending finishes. `Error` is nil on success. (optional)
     public func send(_ mail: Mail, completion: ((Error?) -> Void)? = nil) {
-        send([mail]) { (_, failed) in
+        send([mail], completion:  { (_, failed) in
```
Just curious, were these syntax changes necessary?

sanzaru replied:
Necessary, no - it was just XCode 12.2 (12B45b) complaining with a warning: 
> Backward matching of the unlabeled trailing closure is deprecated; label the argument with 'completion' to suppress this warning

dannys42 replied:
Oh interesting.  I'll make a note to review that [Issue 116](https://github.com/Kitura/Swift-SMTP/issues/116)

dannys42 reviewed (approved):
Looks good.

dannys42 commented:
Would you mind rebasing to master?

sanzaru commented:
Rebase done.

sanzaru commented:
I fixed the tests. While running them, I found out that the line problem also occurs on data and html attachments. Should be also fixed now. All tests were successful, except TestTLSMode.testIgnoreTLS(), but that seems to be because of my server configuration, not because of the test.

lmcd commented:
Is this gonna be merged anytime soon?

dannys42 commented:
Thanks @sanzaru and @mbarnach.  I think there was some travis issue previously.  I'm restarting the build.

dannys42 commented:
I still need to look into what it'll take to get tests working again [Issue 115](https://github.com/Kitura/Swift-SMTP/issues/115)  But I'll push this through in the mean time.
